Source/ThirdParty/skia/CMakeLists.txt

```diff
@@ -0,0 +1,1019 @@
+    src/sksl/transform/SkSLFindAndDeclareBuiltinStructs.cpp
+    src/sksl/transform/SkSLFindAndDeclareBuiltinVariables.cpp
+    src/sksl/transform/SkSLHoistSwitchVarDeclarationsAtTopLevel.cpp
+    src/sksl/transform/SkSLRenamePrivateSymbols.cpp
+    src/sksl/transform/SkSLReplaceConstVarsWithLiterals.cpp
+    src/sksl/transform/SkSLRewriteIndexedSwizzle.cpp
+
+    src/utils/SkCharToGlyphCache.cpp
+    src/utils/SkCustomTypeface.cpp
+    src/utils/SkDashPath.cpp
+    src/utils/SkMatrix22.cpp
+    src/utils/SkNWayCanvas.cpp
+    src/utils/SkPatchUtils.cpp
+    src/utils/SkPolyUtils.cpp
+    src/utils/SkShaderUtils.cpp
+    src/utils/SkShadowTessellator.cpp
+    src/utils/SkShadowUtils.cpp
+
+    src/opts/SkOpts_hsw.cpp
+    src/opts/SkOpts_skx.cpp
+
+    src/ports/SkDebug_stdio.cpp
+    src/ports/SkFontConfigInterface.cpp
+    src/ports/SkFontConfigInterface_direct.cpp
+    src/ports/SkFontConfigInterface_direct_factory.cpp
+    src/ports/SkFontHost_FreeType.cpp
+    src/ports/SkFontHost_FreeType_common.cpp
+    src/ports/SkFontMgr_fontconfig.cpp
+    src/ports/SkGlobalInitialization_default.cpp
+    src/ports/SkImageGenerator_skia.cpp
+    src/ports/SkMemory_malloc.cpp
+    src/ports/SkOSFile_posix.cpp
+    src/ports/SkOSFile_stdio.cpp
+
+    src/sfnt/SkOTTable_name.cpp
+    src/sfnt/SkOTUtils.cpp
+
+    modules/skcms/skcms.cc
+    modules/skcms/src/skcms_TransformBaseline.cc
+    modules/skcms/src/skcms_TransformHsw.cc
+    modules/skcms/src/skcms_TransformSkx.cc
+)
+
+find_package(Epoxy 1.5.4 REQUIRED)
+find_package(Freetype 2.9.0 REQUIRED)
+find_package(Fontconfig 2.13.0 REQUIRED)
+find_package(PNG REQUIRED)
+
+set_target_properties(Skia PROPERTIES
+    CXX_STANDARD 17
+)
+
+target_link_libraries(Skia PRIVATE
+    Epoxy::Epoxy
+    Fontconfig::Fontconfig
+    Freetype::Freetype
+    PNG::PNG
+)
+
+WEBKIT_ADD_TARGET_CXX_FLAGS(Skia
+    -ffp-contract=off
+    -fstrict-aliasing
+
+    -Wno-attributes
+    -Wno-cast-align
+    -Wno-dangling-reference
+    -Wno-deprecated
+    -Wno-psabi
+    -Wno-suggest-attribute=format
+    -Wno-undef
+    -Wno-uninitialized
+    -Wno-unused-parameter
+)
+
+add_custom_target(Skia_headers_symlink
+    COMMAND ${CMAKE_COMMAND} -E create_symlink
+        "${CMAKE_CURRENT_SOURCE_DIR}/include"
+        "${FORWARDING_HEADERS_DIR}/skia"
+    BYPRODUCTS
+        "${FORWARDING_HEADERS_DIR}/skia"
+    COMMENT "Link Skia headers"
+    VERBATIM
+)
+add_dependencies(Skia Skia_headers_symlink)
+target_include_directories(Skia PRIVATE
+    "${CMAKE_CURRENT_SOURCE_DIR}"
+)
+target_include_directories(Skia INTERFACE
+    "${CMAKE_CURRENT_SOURCE_DIR}"
+    "${FORWARDING_HEADERS_DIR}"
+)
+
+target_compile_definitions(Skia PRIVATE
+    SKIA_IMPLEMENTATION=1
+
+    SK_GAMMA_APPLY_TO_A8
+)
+
+target_compile_definitions(Skia PUBLIC
+    SK_DISABLE_LEGACY_IMAGE_READBUFFER
+    SK_DISABLE_LEGACY_INIT_DECODERS
+    SK_DISABLE_LEGACY_PNG_WRITEBUFFER
+    SK_DISABLE_TRACING
+    SK_DISABLE_WEBGL_INTERFACE
+
+    SK_ENABLE_AVX512_OPTS
```

Comment:
Not sure what these options are yet but I see some architecture checks below so should that go there?

Reply:
It's “opts” for “optimizations”; in this case it's safe to always set the macro because there are additional guards _also_ checking for x86 in the relevant source files. I guess it could be moved below in the `if (WTF_CPU_X86 OR WTF_CPU_X86_64)` block.